core/controllers/base.py

```diff
@@ -48,11 +48,13 @@
     'oppia_csrf_secret', {'type': 'unicode'},
     'Text used to encrypt CSRF tokens.', DEFAULT_CSRF_SECRET)
 
-# NOTE: These handlers manage user sessions. Thus, we should never reject or
-# replace them when running in maintenance mode; otherwise admins will be unable
-# to access the site.
+# NOTE: These handlers manage user sessions and serve auth pages. Thus, we
+# should never reject or replace them when running in maintenance mode;
+# otherwise admins will be unable to access the site.
 AUTH_HANDLER_PATHS = (
     '/csrfhandler',
+    '/login',
```

Comment:
Bit concerned about this. Won't it allow people to log in / sign up when the server is in maintenance mode?

Reply:
![image](https://user-images.githubusercontent.com/24855641/127636590-49af4725-3582-4073-a8e1-cd8577be4a05.png)
![image](https://user-images.githubusercontent.com/24855641/127636630-57ddbf64-2dbd-4907-8332-4b5de074b7ce.png)

Aren't we doing the same using build.py

Reply:
Oh, interesting -- thanks for pointing me to that.

Could you get review on this part from @vojtechjelinek and @brianrodri, please? If they approve this then so do I.

No other comments from my end.

Reply:
@vojtechjelinek @brianrodri PTAL

Reply:
I'm not against this change, but why is it needed now and wasn't before?

Reply:
because they are no static pages in `app_dev.yaml` now and neither they are supposed to be there within future.

Reply:
Earlier login and signup routes are registered with the app_dev.yaml, but they are migrated to the angular router now.
They are no `login-page.mainpage.html` or `signup-page.mainpage.html` now. The requests to the /login and /signup routes, go to the oppia-root.mainpage.html and then angular router decides which view to show based on the URL.

Reply:
So, just to be clear -- it is still technically possible for anyone to log in during maintenance mode, correct? They just can't do anything and will see a 503 immediately?

If so, this seems fine -- but I wanted to confirm that it's actually the post-PR behaviour. (Has this been tested manually? Would like to see a video of this in the PR description.)

Reply:
Present behaviour on develop:
Anyone can access the login page but only admins can access the functionality after logging in. Ordinary users will see a maintenance mode immediately after logging in. Maintenance page has a UI bug at present on develop. This PR also fixes that.

https://drive.google.com/file/d/1ZksucmKO7NUunTMLlPnMQMVHKz9EeHSA/view?usp=sharing

Behaviour after this PR gets merged:
https://drive.google.com/file/d/16aa1tiNScvqAs_Vozl5CuGTOkKlVLLmo/view?usp=sharing

Reply:
OK, this seems fine, thanks. Which UI bug are you referring to? There is still the orange message showing on the /login screen for the video after this PR gets merged, so that doesn't seem to be fixed if that's what you're talking about. (I don't think it needs to be fixed in this PR but would you be OK with filing a bug for it?)

Reply:
![Uploading vlcsnap-2021-08-04-23h01m36s070.png…]()

The content of maintenance page is empty.

Reply:
Uploading for @ashutoshc8101 since he is not able to upload to github for some reason:

Behaviour in develop before this PR is merged

![vlcsnap-2021-08-04-23h01m36s070](https://user-images.githubusercontent.com/10575562/128236265-21dcfdab-e3bf-410b-a106-37e34eb3a6d0.png)


Behaviour 
![vlcsnap-2021-08-04-23h07m44s055](https://user-images.githubusercontent.com/10575562/128236334-fd2ad0cb-2be0-49a0-8ac2-54f5876eb0b8.png)
after this PR is merged